# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium/sandbox/win/src/win_utils.h
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 3
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium/sandbox/win/src/win_utils.h
--- a/security/sandbox/chromium/sandbox/win/src/win_utils.h	Wed Jun 12 21:55:59 2019 +1200
+++ b/security/sandbox/chromium/sandbox/win/src/win_utils.h	Wed Jun 12 11:10:48 2019 +0100
@@ -5,11 +5,12 @@
 #ifndef SANDBOX_SRC_WIN_UTILS_H_
 #define SANDBOX_SRC_WIN_UTILS_H_
 
+#include <stddef.h>
 #include <windows.h>
-#include <stddef.h>
 #include <string>
 
 #include "base/macros.h"
+#include "base/stl_util.h"
 #include "base/strings/string16.h"
 #include "sandbox/win/src/nt_internals.h"
 
@@ -17,27 +18,25 @@
 
 // Prefix for path used by NT calls.
 const wchar_t kNTPrefix[] = L"\\??\\";
-const size_t kNTPrefixLen = arraysize(kNTPrefix) - 1;
+const size_t kNTPrefixLen = base::size(kNTPrefix) - 1;
 
 const wchar_t kNTDevicePrefix[] = L"\\Device\\";
-const size_t kNTDevicePrefixLen = arraysize(kNTDevicePrefix) - 1;
+const size_t kNTDevicePrefixLen = base::size(kNTDevicePrefix) - 1;
 
 // Automatically acquires and releases a lock when the object is
 // is destroyed.
 class AutoLock {
  public:
   // Acquires the lock.
-  explicit AutoLock(CRITICAL_SECTION *lock) : lock_(lock) {
+  explicit AutoLock(CRITICAL_SECTION* lock) : lock_(lock) {
     ::EnterCriticalSection(lock);
-  };
+  }
 
   // Releases the lock;
-  ~AutoLock() {
-    ::LeaveCriticalSection(lock_);
-  };
+  ~AutoLock() { ::LeaveCriticalSection(lock_); }
 
  private:
-  CRITICAL_SECTION *lock_;
+  CRITICAL_SECTION* lock_;
   DISALLOW_IMPLICIT_CONSTRUCTORS(AutoLock);
 };
 
@@ -47,8 +46,8 @@
 class SingletonBase {
  public:
   static Derived* GetInstance() {
-    static Derived* instance = NULL;
-    if (NULL == instance) {
+    static Derived* instance = nullptr;
+    if (!instance) {
       instance = new Derived();
       // Microsoft CRT extension. In an exe this this called after
       // winmain returns, in a dll is called in DLL_PROCESS_DETACH
@@ -66,6 +65,15 @@
   }
 };
 
+// Function object which invokes LocalFree on its parameter, which must be
+// a pointer. Can be used to store LocalAlloc pointers in std::unique_ptr:
+//
+// std::unique_ptr<int, sandbox::LocalFreeDeleter> foo_ptr(
+//     static_cast<int*>(LocalAlloc(LMEM_FIXED, sizeof(int))));
+struct LocalFreeDeleter {
+  inline void operator()(void* ptr) const { ::LocalFree(ptr); }
+};
+
 // Convert a short path (C:\path~1 or \\??\\c:\path~1) to the long version of
 // the path. If the path is not a valid filesystem path, the function returns
 // false and argument is not modified.
@@ -94,7 +102,7 @@
 
 // Translates a reserved key name to its handle.
 // For example "HKEY_LOCAL_MACHINE" returns HKEY_LOCAL_MACHINE.
-// Returns NULL if the name does not represent any reserved key name.
+// Returns nullptr if the name does not represent any reserved key name.
 HKEY GetReservedKeyFromName(const base::string16& name);
 
 // Resolves a user-readable registry path to a system-readable registry path.
@@ -106,9 +114,10 @@
 // Writes |length| bytes from the provided |buffer| into the address space of
 // |child_process|, at the specified |address|, preserving the original write
 // protection attributes. Returns true on success.
-bool WriteProtectedChildMemory(HANDLE child_process, void* address,
-                               const void* buffer, size_t length,
-                               DWORD writeProtection = PAGE_WRITECOPY);
+bool WriteProtectedChildMemory(HANDLE child_process,
+                               void* address,
+                               const void* buffer,
+                               size_t length);
 
 // Returns true if the provided path points to a pipe.
 bool IsPipe(const base::string16& path);